### DISCUSSION TITLE
Ability to create custom file viewer

### DISCUSSION THREAD

**Discussion by bebef1987** [community/community#15132] [Other Feature Feedback, Questions, & Ideas]
From:  #https://github.com/isaacs/github/issues/1005
"
Add-ons for Bitbucket cloud can register [file viewer  #modules](https://developer.atlassian.com/bitbucket/modules/fileViews.html). These file viewers make Bitbucket send the viewed source of a file in a repository to the add-on's server, where it can be rendered into a format more suitable for human consumption, after which it is shown on Bitbuckets file explorer. A nice example of an add-on using this feature is the [FileViewer add-on](https://blog.bitbucket.org/2015/12/29/file-viewer-for-bitbucket/).

These kind of file viewers are very handy when working with repositories containing e.g CAD files or special image formats not supported by default. Having these files rendered in e.g a pull request allows for awesome improvements in collaborative workflow, especially since people dealing with these kind of files are typically not that familiar with git workflows, causing them to rely more on web UIs.

Github does not have support for custom file viewers. It would be awesome if it did.
"

I still see value in creating a custom file viewer for some file types. 
One example would be building a custom .xaml viewer

**Comment by timonsku** [comment]
This would be so cool. Especially for open source hardware work where you deal with a lot of CAD besides code.
Even a client side renderer that you can define for your repo would do the job. Similar to how github action packages work you could define a repo as file viewer addon that renders a certain file type client side and can be linked into your repo if relevant to your files.

**Comment by notluquis** [comment]
Another concrete use case for this.

Databricks notebooks in "source format" are plain `.py` files that use `# COMMAND ----------` as the cell separator and `# MAGIC %md` for markdown cells. GitHub renders them as an undifferentiated Python file, so documentation cells show up as commented-out noise and there is no rich diff in pull requests.

`.ipynb` already gets a rich view, and this is the same notebook structure in a different encoding, so a repo-declared viewer, or even just a way to map a file glob onto an existing renderer, would cover it without GitHub having to support the format itself.

Editors already solve this locally with a single setting: VS Code accepts a `jupyter.interactiveWindow.cellMarker.codeRegex` regex and renders the cells correctly. Something equivalent declared in the repo would be enough.
